pkg/skaffold/runner/verify.go

```diff
@@ -89,15 +89,8 @@ See https://skaffold.dev/docs/pipeline-stages/taggers/#how-tagging-works`)
 	ctx, endTrace := instrumentation.StartTrace(ctx, "Verify_Verifying")
 	defer endTrace()
 
-	// we only want to register images that are local AND were built by this runner OR forced to load via flag
-	var localAndBuiltImages []graph.Artifact
-	for _, image := range localImages {
-		if r.runCtx.ForceLoadImages() || r.wasBuilt(image.Tag) {
-			localAndBuiltImages = append(localAndBuiltImages, image)
-		}
-	}
-
-	r.verifier.RegisterLocalImages(localAndBuiltImages)
+	// We assume all the localImgs were build by Skaffold, so we want to load them all.
+	r.verifier.RegisterLocalImages(localImages)
```

Comment:
Hey @ericzzzzzzz, thanks for going deeper wit the load images logic. From our offline conversation:
- I created a new issue https://github.com/GoogleContainerTools/skaffold/issues/8946 to follow-up with the improvement for the load image logic, mentioning the points you added in the previous comment; feel free to add/modify any more details if need it
- Currently, for the `verify` command, the [`artifacts` variable](https://github.com/GoogleContainerTools/skaffold/blob/453b101d70ccbf75c98f945516589950c87f63a0/pkg/skaffold/runner/verify.go#L54) that is used to calculate the images to load in the clusters, is populated with the resulting values from the intersection between:
  -  the images found in all the verify test cases (filtered by profile), and 
  -  the images received in the `build-artifacts` flag, 
  -  we don't take into account the values from the `build` stanza.
- As discussed, I added back the logic that calculates the images to load in the clusters, and instead now we are turning on the `--load-images` flag, which will allow users to force the loading of the images in the cluster

Please let me know if you have more comments :)

Reply:
Hey @renzodavid9 ,
 - the `verify` command uses `imageLoader` to load the image under the hood, the imageLoader take the intersection of localImage, originalImage, allBuilds
  - The local image slice passed by k8s verifier is from https://github.com/GoogleContainerTools/skaffold/blob/73a798a280e46d9ddaddc26d9caf05c268759d33/pkg/skaffold/runner/verify.go#L92C1-L100
  - The source of original images is from Build Stanza  in skaffold yaml file. 
  - allBuilds is the one you mentioned, -- the intersection of imaged defined in verify stanza and build-output from command line
- You can test with your branch to see if verify command if taking build stanza into account when loading image by the following steps:
  - create a temp folder
  - create a Dockerfile with following content
    ```Dockerfile
    FROM ubuntu as starter
    ```
   - create skaffold yaml file
   ```yaml
  apiVersion: skaffold/v4beta6
  kind: Config
  build:
  artifacts:
    - image: verify-starter
      context: .
  verify:
    - name: vvv
      container:
        name: vvvaaa
        image: verify-starter
        command: [ "/bin/sh" ]
        args: [ "-c", "echo 123; sleep 1; echo bye" ]
      executionMode:
        kubernetesCluster: {}
   ```
   - run ``skaffold build -q | skaffold build -q > build_result.json ``
   - delete build stanza
   - run ``kind delete cluster && kind create cluster`` to make sure we use a brand new cluster to test.
   - run ``skaffold verify -a build_result.json --load-images``, this will fail due to image doesn't exit on cluster 
- I'm thinking we just change here https://github.com/GoogleContainerTools/skaffold/blob/e50818aaf6cf61b9da9bc5e4acb06837eabee2dd/pkg/skaffold/verify/k8sjob/verify.go#L138 to  `v.imageLoader.LoadImages(childCtx, out, allbuilds, allbuilds, allbuilds);`, this way we don't need to change the flag, also the intersection of image under verify and from cli will be loaded. WDYT?

Reply:
Oh, I see, good catch. I modified the call to `imageLoader` in https://github.com/GoogleContainerTools/skaffold/blob/e50818aaf6cf61b9da9bc5e4acb06837eabee2dd/pkg/skaffold/verify/k8sjob/verify.go#L138 to `v.imageLoader.LoadImages(childCtx, out, v.localImages, v.localImages, v.localImages);` I'm using here `v.localImages` instead of `allbuilds` to be able to control if the images should be loaded into Kind/K3D using [Skaffold's global configuration file](https://skaffold.dev/docs/design/global-config/), setting the `k3d-disable-load` and `kind-disable-load` properties. Let me know what do you think. Thanks!

Reply:
LGTM